src/main/java/com/google/flourbot/entity/EntityModule.java

```diff
@@ -0,0 +1,86 @@
+package com.google.flourbot.entity;
+
+import com.google.flourbot.entity.EntityInterface;
+import com.google.flourbot.datastorage.FirebaseDataStorage;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.flourbot.entity.action.Action;
+import com.google.flourbot.entity.action.SheetAppendAction;
+import com.google.flourbot.entity.trigger.Trigger;
+import com.google.flourbot.entity.trigger.CommandTrigger;
+import java.util.Map;
+
+import java.lang.Exception;
+
+public class EntityModule implements EntityInterface {
+    
+    private static EntityModule instance = null;
+    private final FirebaseDataStorage datastorage;
+
+    private EntityModule() {
+        this.datastorage = new FirebaseDataStorage();
+    }
+
+    public Macro getMacro(String userEmail, String macroName) throws Exception {
+        QueryDocumentSnapshot document = this.datastorage.getDocument(userEmail, macroName); 
+        //TODO 
+        //Handle Java "Optional<QueryDocumentSnapshot>" once getDocument function is modified" 
+
+        Map<String, Object> macroMap = document.getData();
+        Map<String, Object> triggerMap = (Map<String, Object>) macroMap.get("trigger");
+        Map<String, Object> actionMap = (Map<String, Object>) macroMap.get("action");
+
+        String creatorId = (String) macroMap.get("creatorId");
+        Trigger trigger = getTrigger(triggerMap);
+        Action action = getAction(actionMap);
+
+        Macro macro = new Macro(creatorId, macroName, trigger, action);
+
+        return macro;
+    }
+
+    private Trigger getTrigger(Map<String, Object> triggerMap) throws Exception {
+
+        Trigger trigger;        
+        String triggerType = (String) triggerMap.get("type");
+
+        switch(triggerType) {
+            case ("Command Trigger"): 
```

Comment:
nit, enum would be perfect, but if we cannot have enum we should use the same name standard so "COMMAND_TRIGGER"

Reply:
Noted! This require some code change on the client side as well.

Reply:
Issue #14